sdk/mixedreality/azure-mixedreality-authentication/azure/mixedreality/authentication/_utils.py

```diff
@@ -0,0 +1,73 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+import base64
+import json
+import random
+
+from azure.core.credentials import AccessToken
+
+from ._generated.models import StsTokenResponseMessage
+
+
+def convert_to_access_token(token_response_message):
+    # type: (StsTokenResponseMessage) -> AccessToken
+    """
+    Converts the specified token response message to an AccessToken.
+    """
+    if not StsTokenResponseMessage:
+        raise ValueError("token_response_message can not be None")
+
+    expiration_timestamp = retrieve_jwt_expiration_timestamp(token_response_message.access_token)
+
+    return AccessToken(token_response_message.access_token, expiration_timestamp)
+
+def retrieve_jwt_expiration_timestamp(jwt_value):
+    # type: (str) -> int
+    """
+    Retrieves the expiration value from the JWT.
+
+    :param str jwt_value: The JWT value.
+    :returns: int
+    """
+    if not jwt_value:
+        raise ValueError("jwt_value can not be None")
+
+    parts = jwt_value.split(".")
+
+    if len(parts) < 3:
+        raise ValueError("Invalid JWT structure.")
```

Comment:
This may be deliberate, but this error gives no indication what the problem is. Perhaps: "Invalid JWT structure. Expected blah format."

Reply:
Thanks. Yes. I've added some more specific information.